vhdbuilder/packer/backfill-cleanup.sh

```diff
@@ -49,28 +49,46 @@ fi
 if [[ -n "${AZURE_RESOURCE_GROUP_NAME}" ]]; then
   gallery_list=$(az sig list -g ${AZURE_RESOURCE_GROUP_NAME} | jq -r '.[] | select(.name != "AKSWindows") | .name')
   for gallery in $gallery_list; do
-    # delete old Windows SIG image versions in gallery (image definitions must have .osType == "Windows")
-    image_defs=$(az sig image-definition list -g ${AZURE_RESOURCE_GROUP_NAME} -r ${gallery} | jq -r '.[] | select(.osType == "Windows").name')
-    for image_definition in $image_defs; do
-        echo "Finding sig image versions associated with ${image_definition} in gallery ${gallery}"
-        image_versions=$(az sig image-version list -g ${AZURE_RESOURCE_GROUP_NAME} -r ${gallery} -i ${image_definition} | jq --arg dl $deadline -r '.[] | select(.tags.now < $dl).name')
-        for image_version in $image_versions; do
-            az sig image-version show -e $image_version -i ${image_definition} -r ${gallery} -g ${AZURE_RESOURCE_GROUP_NAME} | jq .id
-            echo "Deleting sig image-version ${image_version} ${image_definition} from gallery ${gallery} rg ${AZURE_RESOURCE_GROUP_NAME}"
-            az sig image-version delete -e $image_version -i ${image_definition} -r ${gallery} -g ${AZURE_RESOURCE_GROUP_NAME}
-        done
-        image_versions=$(az sig image-version list -g ${AZURE_RESOURCE_GROUP_NAME} -r ${gallery} -i ${image_definition})
-        # clean the image-definition if ALL associated sig versions have been deleted
-        if [[ "$image_versions" == "[]" ]]; then
-          echo "Deleting sig image-definition ${image_definition} from gallery ${gallery} rg ${AZURE_RESOURCE_GROUP_NAME}"
-          az sig image-definition delete --gallery-image-definition ${image_definition} -r ${gallery} -g ${AZURE_RESOURCE_GROUP_NAME}
-        fi
-    done
-    image_defs=$(az sig image-definition list -g ${AZURE_RESOURCE_GROUP_NAME} -r ${gallery} | jq -r '.[] | select(.osType == "Windows").name')
-    # clean the gallery if ALL sig image-definitions have been deleted
-    if [[ -z $image_defs ]]; then
-      echo "Deleting gallery ${gallery}"
-      az sig delete --gallery-name ${gallery} --resource-group ${AZURE_RESOURCE_GROUP_NAME}
+    if [[ "${gallery}" =~ WS2019Gallery* ]]; then
+      create_date=${gallery:13:6}
+    elif [[ "${gallery}" =~ WS2019_containerdGallery* ]]; then
+      create_date=${gallery:24:6}
+    elif [[ "${gallery}" =~ WS2022_containerdGallery* ]]; then
+      create_date=${gallery:24:6}
+    elif [[ "${gallery}" =~ WS2022_containerd_gen2Gallery* ]]; then
+      create_date=${gallery:29:6}
+    else
+      continue
+    fi
+
+    due_date=$(date +%y%m%d -d "7 days ago")
+    echo "create_date is ${create_date}"
+    echo "due_date is ${due_date}"
+    # clean the entire SIG resources if it's one week ago
+    if [[ $create_date < $due_date ]]; then
+      echo "Finding sig image definitions from gallery ${gallery}"
```

Comment:
adding back old logic because running commands cannot delete nested resources unlike the portal.

The experiments - I created a bunch of sig image defs, versions.

![image](https://user-images.githubusercontent.com/103480804/221911402-2c39da5b-c568-42fc-9630-17f8146bed15.png)

dry-run of line `71` to line `91` locally showing that the bottom-up deletion succeeded.
![image](https://user-images.githubusercontent.com/103480804/221911627-62b6ee2e-1ffa-477c-aca7-29cf99529ac4.png)

Reply:
The proof of the dry-run test will be posted on the channel. (showing that this logic works)